databases/relational-databases.md

```diff
@@ -39,69 +76,132 @@ To understand, query, and insert information into a relational database, a techn
+  - DROP – is used to delete objects from the database.
+  - ALTER-is used to alter the structure of the database.
+  - TRUNCATE–is used to remove all records from a table, including all spaces allocated for the records are removed.
+  - COMMENT –is used to add comments to the data dictionary.
+  - RENAME –is used to rename an object existing in the database.
+
 - DDL to create tables
   - `CREATE TABLE`
-- Bonus topic for the advanced: foreign key constraints
-- Explain different databases with slightly different dialects (MySQL - PostgreSQL)
-- Explain existence of other types of non-relational, non-sql databases that will be encountered
-  - examples key-value (redis), graph (Neo4j), streaming (cassandra), document (mongo)
 
-### Materials
+- Adavntages of Relational Database:
+  - Relational databases are critical to moving, sharing and maintaining the integrity of data in the modern workplace where users working with PCs, workstations, and mainframes all need access to the same data. Using an RDBMS offers the following advantages:
+     - Speed. True relational databases support client server architecture. Populate times for large ODBC databases will not grow as quickly as populate times for Paradox databases.
+     - Structured Query Language (SQL), which is the basis for relational database management systems. SQL is the standard data access language of the American National Standards Institute (ANSI) and the International Standards Organization (ISO). If you are familiar with SQL, you can access any RDBMS.
+     - Incremental data storage, which gives you a historical perspective of the data.
+     - Client/Server configuration, which lets you run Visualizer on one machine while storing data on another machine.
+     - A single interface, which provides integrated data that can be shared across platforms, networks, and devices.
+- Describing an ORM:
+  - Object-relational mapping (ORM) is a programming technique in which a metadata descriptor is used to connect object code to a relational database. Object code is written in object-oriented programming (OOP) languages such as Java or C#. ORM converts data between type systems that are unable to coexist within relational databases and OOP languages.
 
-#### Education
+- Bonus topic for the advanced: foreign key constraints
+  - A FOREIGN KEY is a key used to link two tables together.
+  - A FOREIGN KEY is a field (or collection of fields) in one table that refers to the PRIMARY KEY in another table.
+  - Below is an example of its usage:
+    ```sql
+      CREATE TABLE Orders (
+      OrderID int NOT NULL,
+      OrderNumber int NOT NULL,
+      PersonID int,
+      PRIMARY KEY (OrderID),
+      FOREIGN KEY (PersonID) REFERENCES Persons(PersonID);
+    ```
+The table containing the foreign key is called the child table, and the table containing the candidate key is called the referenced or parent table.
```

Comment:
https://en.wikipedia.org/wiki/Foreign_key

Reply:
do I need to add this link?

Reply:
YES